### PR TITLE
fix: improper token revocation when logging out and showAccessToken=true

### LINKED ISSUE #1095: Improper token revocation when logging out and `showAccessToken=true`
### Issue and Steps to Reproduce

Use `packages/vanilla-demo`

Set `OidcTrustedDomains.js` to:
```js
const trustedDomains = {
  'my-config': { // do NOT try to reproduce it on `default`, tokens on `default` won't be shown because of another bug: https://github.com/AxaFrance/react-oidc/pull/1094
    domains: ['*'],
    showAccessToken: true, // <-- this matters
  },
};
```

In `index.tsx`, change:

```ts
const vanillaOidc = VanillaOidc.getOrCreate(() => fetch)(configuration);
```

to

```ts
const vanillaOidc = VanillaOidc.getOrCreate(() => fetch)(configuration, 'my-config');
```

### Versions

6.24.1, the most recent version at the moment

### Screenshots

image

#### Expected

No errors during revocation

#### Actual

An exception is thrown, `Response is undefined inside a success` is logged in the console, the revocation request is not sent.

### Additional Details

If `showAccessToken` is `true`, then the request body contains the actual access token.

If `showAccessToken` is `false`, then the token in the request body is replaced with `ACCESS_TOKEN_SECURED_BY_OIDC_SERVICE_WORKER`.

Therefore, if `showAccessToken` is `true`, this condition is false:
https://github.com/AxaFrance/react-oidc/blob/3e535c94dc57d59956498bfa0ca9536106039bda/packages/react/service_worker/OidcServiceWorker.ts#L160
which leads to `undefined` being returned:
https://github.com/AxaFrance/react-oidc/blob/3e535c94dc57d59956498bfa0ca9536106039bda/packages/react/service_worker/OidcServiceWorker.ts#L242

It looks like we've got to check for something else, not just `ACCESS_TOKEN_SECURED_BY_OIDC_SERVICE_WORKER`, as it's not present when tokens are not hidden.

### PR DESCRIPTION
**Pull request by kosciolek**
fixes #1095

### CODE REVIEW

File: packages/react/service_worker/OidcServiceWorker.ts [hunk 1]
```diff
@@ -239,24 +239,30 @@ const handleFetch = async (event: FetchEvent) => {
...
+          cache: clonedRequest.cache,
+          redirect: clonedRequest.redirect,
+          referrer: clonedRequest.referrer,
+          credentials: clonedRequest.credentials,
+          integrity: clonedRequest.integrity,
+        });
       });
       response
         .then((r) => {
-          if (r !== undefined) {
```
Comment (kosciolek) [inline packages/react/service_worker/OidcServiceWorker.ts:246]: The response no longer can be `undefined`, so I think we can just remove this

File: packages/react/service_worker/OidcServiceWorker.ts [hunk 2]
```diff
@@ -239,24 +239,30 @@ const handleFetch = async (event: FetchEvent) => {
             integrity: clonedRequest.integrity,
           }).then(hideTokens(currentDatabase));
         }
-        return undefined;
+
```
Comment (kosciolek) [inline packages/react/service_worker/OidcServiceWorker.ts:242]: **Is there a reason why this `undefined` was here previously?**

I think this is the simplest way to fix the issue, but I don't want to break something else

File: packages/react/service_worker/OidcServiceWorker.ts [hunk 3]
```diff
@@ -239,24 +239,30 @@ const handleFetch = async (event: FetchEvent) => {
             integrity: clonedRequest.integrity,
           }).then(hideTokens(currentDatabase));
         }
-        return undefined;
```
[the remark this answers is not in the document]
Comment (kosciolek) [reply packages/react/service_worker/OidcServiceWorker.ts:242]: Sure! We're waiting for this patch to make a release on our side though, any idea when the big PR might get merged?

[the remark this answers is not in the document]
Comment (guillaume-chervet) [reply packages/react/service_worker/OidcServiceWorker.ts:242]: hi @kosciolek , The big PullRequest has been merged :)

[the remark this answers is not in the document]
Comment (guillaume-chervet) [reply packages/react/service_worker/OidcServiceWorker.ts]: thank you @kosciolek. I test it !

**Review by guillaume-chervet — APPROVED** [review]
Thank you @kosciolek .
It works perfectly.

**Pull request merged**